atlasdb-tests-shared/src/main/java/com/palantir/atlasdb/sweep/AbstractSweepTest.java

```diff
@@ -86,22 +91,33 @@ protected CacheWarming getSsmCacheWarming() {
         return CacheWarming.FULL;
     }
 
+    @Rule
+    public TemporaryFolder tempFolder = new TemporaryFolder();
+
     @Before
     public void setup() {
         kvs = kvsManager.getDefaultKvs();
         ssm = SweepStrategyManagers.create(kvs, getSsmCacheWarming());
-        txManager = getManager();
+        services = InMemoryTimelockServices.create(tempFolder);
+        txManager = createAndRegisterManager();
         txService = TransactionServices.createRaw(kvs, txManager.getTimestampService(), false);
         SweepTestUtils.setupTables(kvs);
     }
 
+    @After
+    public void tearDown() {
+        if (services != null) {
+            services.close();
+        }
```

Comment:
I can probably remove this now. If you mess something up in initialising your `InMemoryTimelockService`, then `services` will end up null, but we'll still call the `@After` method, which will add an NPE to the actual exception.

(In particular, I was getting issues with "Database busy" errors when two things were creating TimelockAgents at once)

Reply:
Aaaah, those are flakes when you try and create two SQLite databases at the same time.

You can solve that with sequential execution of tests, although that does slow them down. That, or work out a fix such that they don't try and contend on resources.

I'd strongly recommend _guaranteeing_ that these are fixed so that we don't get flakes down the line.

Reply:
I mean, ideally we wouldn't really need an SQLite database at all here, but I suppose we want to keep this close to production.

The original fix here was to use a different data directory for each test, so different unit tests in the same class wouldn't contend with each other - that's why the `close()` method exists. Particular tests were failing (Cassandra integration tests) because some static setup code was creating a `TimeLockAgent`, and then the per-test setup code was also creating a `TimeLockAgent`. Fixing that is a job for another PR.

Reply:
We definitely do want to use SQLite to keep as close to prod as possible (as you note).

So long as the changes here don't introduce further flakes, I'm happy.